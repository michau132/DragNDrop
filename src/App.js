import React from 'react';
import styled from 'styled-components';
import {
  Paper, Grid,
} from '@material-ui/core';
import Map from './Map';
import Input from './FileInput';
import FileList from './FileList';
import Dropzone from './DropZone';
import GlobalStyle from './GlobalStyle';
import SubmitFile from './SubmitFile';

const Main = styled.main`
  display: flex;
  justify-content:center;
  align-items: center;
`;

const PaperWithPadding = styled(Paper)`
  && {
  padding: 10px;
  background-color: rgba(178,161,161, 0.6);
  width: 100%;
  }
`;

const GridWithNoMargin = styled(Grid)`
  && {
    margin: 0;
    width: 100%;
  }
`;

class App extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      images: [],
    };
  }

  componentDidMount() {
    if (!('draggable' in document.createElement('span')) || !window.FileReader) {
      // eslint-disable-next-line no-alert
      return alert('Your browser does not support dragging');
    }
    return null;
  }

  updateImages = (item) => {
    this.setState(prevState => ({
      images: [...prevState.images, item],
    }));
  }

  removeImage = (index) => {
    this.setState((prevState) => {
      const images = prevState.images.filter((_, i) => index !== i);
      return {
        images,
      };
    });
  }

  render() {
    const { images } = this.state;
    return (
      <Main>
        <GlobalStyle />
        <PaperWithPadding>
          <Map images={images} />
          <Paper>
            <GridWithNoMargin container justify="space-around" spacing={40}>
              <SubmitFile
                images={images}
                updateImages={this.updateImages}
                render={props => (
                  <Grid item md={4} container justify="center">
                    <Input {...props} />
                    <Dropzone {...props} />
                  </Grid>
                )}
              />
              <Grid item md={8}>
                <FileList
                  list={images}
                  removeImage={this.removeImage}
                />
              </Grid>
            </GridWithNoMargin>
          </Paper>
        </PaperWithPadding>
      </Main>
    );
  }
}

export default App;
