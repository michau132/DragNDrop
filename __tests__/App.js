import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/App';


describe('<App/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should update state', () => {
    wrapper.instance().updateImages({ avatar: 'dd', pos: { lng: 12.12, lat: 12.13 } });
    expect(wrapper.state().images[0]).toEqual({ avatar: 'dd', pos: { lng: 12.12, lat: 12.13 } });

    wrapper.instance().removeImage(0);
    expect(wrapper.state().images).toEqual([]);
  });
});
