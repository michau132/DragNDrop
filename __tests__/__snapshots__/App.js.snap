// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match snapshot 1`] = `
<styled.main>
  <GlobalStyleComponent />
  <Styled(WithStyles(Paper))>
    <Map
      images={Array []}
    />
    <WithStyles(Paper)>
      <Styled(WithStyles(Grid))
        container={true}
        justify="space-around"
        spacing={40}
      >
        <MySnackProvider
          images={Array []}
          render={[Function]}
          updateImages={[Function]}
        />
        <WithStyles(Grid)
          item={true}
          md={8}
        >
          <FileList
            list={Array []}
            removeImage={[Function]}
          />
        </WithStyles(Grid)>
      </Styled(WithStyles(Grid))>
    </WithStyles(Paper)>
  </Styled(WithStyles(Paper))>
</styled.main>
`;
